Issues with MAPL 3 and ifx 2024.2

Well, unlike MAPL 2 (see #2880), MAPL 3 doesn't even build. It gets close, but dies in *linking* GeomIO:
```
[ 88%] Linking Fortran static library ../lib/libMAPL.GeomIO.a
cd /home/mathomp4/Models/MAPL3-IFX-2024.2/build-Debug/GeomIO && /ford1/share/gmao_SIteam/Utilities/cmake-3.28.3/bin/cmake -P CMakeFiles/MAPL.GeomIO.dir/cmake_clean_target.cmake
cd /home/mathomp4/Models/MAPL3-IFX-2024.2/build-Debug/GeomIO && /ford1/share/gmao_SIteam/Utilities/cmake-3.28.3/bin/cmake -E remove /home/mathomp4/Models/MAPL3-IFX-2024.2/build-Debug/lib/libMAPL.GeomIO.a
cd /home/mathomp4/Models/MAPL3-IFX-2024.2/build-Debug/GeomIO && /ford1/share/gmao_SIteam/Utilities/cmake-3.28.3/bin/cmake -E cmake_link_script CMakeFiles/MAPL.GeomIO.dir/link.txt --verbose=1
/usr/bin/ar qc ../lib/libMAPL.GeomIO.a CMakeFiles/MAPL.GeomIO.dir/GeomIO.F90.o CMakeFiles/MAPL.GeomIO.dir/SharedIO.F90.o CMakeFiles/MAPL.GeomIO.dir/Geom_PFIO.F90.o CMakeFiles/MAPL.GeomIO.dir/Grid_PFIO.F90.o CMakeFiles/MAPL.GeomIO.dir/GeomCatagorizer.F90.o CMakeFiles/MAPL.GeomIO.dir/pFIOServerBounds.F90.o
/usr/bin/ar: Relink `/ford1/share/gmao_SIteam/intel/oneapi/compiler/2024.2/lib/libimf.so' with `/lib64/libm.so.6' for IFUNC symbol `cosf'
Error running link command: Segmentation fault
make[2]: *** [GeomIO/CMakeFiles/MAPL.GeomIO.dir/build.make:173: lib/libMAPL.GeomIO.a] Error 1```

I'll mention @bena-nasa as he wrote the layer, but honestly, this error sort of baffles me. GeomIO's CMake is pretty boring and I don't think the code is that spectacularly different from the rest of MAPL3. 

I mean, the error sort of points to a cosine function (I'd imagine) but there is no `cos()` in there. 

I'm going to invoke the name of @climbfuji as maybe he's seen this as he looks at `ifx` in spack-stack land?

--- a/GeomIO/CMakeLists.txt
+++ b/GeomIO/CMakeLists.txt
@@ -12,7 +12,7 @@ set(srcs
 esma_add_library(${this}
   SRCS ${srcs}
   DEPENDENCIES MAPL.geom_mgr MAPL.pfio MAPL.base MAPL.shared MAPL.hconfig_utils GFTL::gftl-v2
-  TYPE ${MAPL_LIBRARY_TYPE}
+  TYPE SHARED
   )
 
 target_include_directories (${this} PUBLIC

